actionview/lib/action_view/cache_expiry.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ActionView
+  class CacheExpiry
+    class Executor
+      def initialize(watcher:)
+        @cache_expiry = CacheExpiry.new(watcher: watcher)
+      end
+
+      def before(target)
+        @cache_expiry.clear_cache_if_necessary
+      end
+    end
+
+    def initialize(watcher:)
+      @watched_dirs = nil
+      @watcher_class = watcher
+      @checker = nil
+    end
+
+    def clear_cache_if_necessary
+      fs_paths = all_view_paths.select do |view_path|
+        view_path.is_a?(FileSystemResolver)
+      end
```

Comment:
Would `grep(FileSystemResolver)` work here?

Also do we need the `uniq` in `all_view_paths` when we're doing it here after sorting? What's quicker / more memory efficient?

Same goes for `sort.uniq` versus `uniq.sort`?

Reply:
👍 Switched to `grep(FileSystemResolver)`.

I removed the `uniq` from `all_view_paths`, because it was unnecessary if we do it here. We need a `uniq` after mapping to the paths because, strangely, `ActionController::Base` has a different Resolver objects than `ActionController::API` (something I'd like to fix, but is outside the scope of this PR).

The paths here are the top-level paths (ie. `app/views`). I don't think there will ever be enough path here that `sort.uniq` vs `uniq.sort` will have any measurable performance difference (there would be a bottleneck in partial rendering before being noticeable here)